collection-scripts/gather_crio_profiling

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# safeguards
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# TODO: check which is the right "BASE_COLLECTION_PATH" to use
+BASE_COLLECTION_PATH="must-gather"
+CRIO_LOG_PATH="${BASE_COLLECTION_PATH}/pprof/crio"
+CRIO_PROFILING_DURATION_SEC=30
+
+mkdir -p "$CRIO_LOG_PATH"
+
+# NOTE: followed gather_core_dumps script
+
+_gather_crio_profiling_data_from_node() {
+	local node="$1"
+    local tmpdir="/tmp/crio_pprof"
+    local curl_pprof_ep="curl --unix-socket /host/var/run/crio/crio.sock http://localhost/debug/pprof/"
+
+    local cp_timeout=$((CRIO_PROFILING_DURATION_SEC+120)) # debug pod timeout waiting for data to be copied
+    local cp_retry_num=5 # how many times we will try to copy the data if we fail
+    local cp_retry_interval=10 # how many seconds to wait before retrying copy data
+
+    # Get debug pod's name
+	debugPod=$(oc debug --to-namespace=default node/${node} -o jsonpath='{.metadata.name}')
+    if [ -z "$debugPod" ]
+    then
+        echo "ERROR: cannot start debug pod for node ${node}"
+        return
+    fi
+	echo "INFO: starting CRI-O profiling on node ${node}"
+    # Start debug pod collecting CRI-O profiling data in the "default" namespace and let it wait we
+    # copy from the pod the profiling data.
+    oc debug --to-namespace="default" node/${node} -- /bin/bash -c "\
+        mkdir -p ${tmpdir} \
+        && ${curl_pprof_ep}profile?seconds=${CRIO_PROFILING_DURATION_SEC} > ${tmpdir}/prof.out \
+        && ${curl_pprof_ep}heap > ${tmpdir}/heap.out \
+        && sleep ${cp_timeout} \
+        " > /dev/null 2>&1 &
+
+    # Data will be ready only after profiling is over.
+    sleep ${CRIO_PROFILING_DURATION_SEC}
+    for i in $(seq 1 $cp_retry_num)
+    do
+        sleep ${cp_retry_interval}
+        oc cp -n default ${debugPod}:${tmpdir}/heap.out ${CRIO_LOG_PATH}/${node}_heap.out > /dev/null 2>&1
+        # file not there? retry!
+        [ -f "${CRIO_LOG_PATH}/${node}_heap.out" ] || continue
+        oc cp -n default ${debugPod}:${tmpdir}/prof.out ${CRIO_LOG_PATH}/${node}_prof.out > /dev/null 2>&1
+        break
+    done
+
+    [ -f "${CRIO_LOG_PATH}/${node}_prof.out" -a -f "${CRIO_LOG_PATH}/${node}_heap.out" ] || \
+        echo "ERROR: cannot retrieve profiling data for node ${node}"
+
+    oc delete pod "$debugPod" -n "default" > /dev/null 2>&1
+}
+
+# get crio profiling data on all nodes in parallel
+gather_crio_profiling_data() {
+    local pids=()
+
+    echo "INFO: starting CRI-O profiling data gathering"
+    for NODE in ${NODES}; do
+        _gather_crio_profiling_data_from_node "${NODE}" &
+        pids+=($!)
+    done
+
+    wait ${pids[@]}
+    echo "INFO: CRI-O profiling data gathering completed"
+}
+
+
+NODES="${*:-$(oc get nodes -o jsonpath='{.items[?(@.status.nodeInfo.operatingSystem=="linux")].metadata.name}')}"
```

Comment:
Well, I understand that there maybe some concerns to collect profiling data for all the nodes in the cluster. Do we really need profiling data from all of them? What issues can this bring?
To be honest, timing doesn't look to be an issue related to the number of node we run onto: the profiling job is run in parallel in each node. So, there is no time increase due to the number of nodes we run this on.
Moreover, we evaluated the impact of running profiling tasks on CRI-O: it is not resource sensitive, so should be really barely (or even not) noticeable.

Reply:
The profiling script is not anymore linked to the main gather script: it should be called directly now (i.e.: oc adm must-gather -- gather_profiling). Probably makes more sense if the user calling the script will issue the command against specific nodes, anyway running on all the nodes in the cluster should be allowed... do you thing we need to add a warning if no nodes are specified (so that we will collect profiling info from all the nodes in the cluster?